quarto/_extensions/countdown/countdown.lua

```diff
@@ -0,0 +1,353 @@
+-- Specify embedded version 
+local countdownEmbeddedVersion = "0.0.1"
+
+-- Only embed resources once if there are multiple timers present 
+local needsToExportDependencies = true
+
+-- List CSS default options
+local default_style = {
+  -- Font size for the countdown element
+  font_size = "3rem",
+  -- Margin around the countdown element
+  margin = "0.6em",
+  -- Padding within the countdown element
+  padding = "10px 15px",
+  -- Shadow applied to the countdown element
+  box_shadow = "0px 4px 10px 0px rgba(50, 50, 50, 0.4)",
+  -- Border width of the countdown element
+  border_width = "0.1875rem",
+  -- Border radius of the countdown element
+  border_radius = "0.9rem",
+  -- Line height of the countdown element
+  line_height = "1",
+  -- Border color of the countdown element
+  color_border = "#ddd",
+  -- Background color of the countdown element
+  color_background = "inherit",
+  -- Text color of the countdown element
+  color_text = "inherit",
+  -- Background color when the countdown is running
+  color_running_background = "#43AC6A",
+  -- Border color when the countdown is running
+  color_running_border = "#2A9B59FF", -- Needs color_darken()
+  -- Text color when the countdown is running
+  color_running_text = 'inherit',
+  -- Background color when the countdown is finished
+  color_finished_background = "#F04124",
+  -- Border color when the countdown is finished
+  color_finished_border = "#DE3000FF",  -- Needs color_darken()
+  -- Text color when the countdown is finished
+  color_finished_text = 'inherit',
+  -- Background color when the countdown has a warning
+  color_warning_background = "#E6C229",
+  -- Border color when the countdown has a warning
+  color_warning_border = "#CEAC04FF", -- Needs color_darken()
+  -- Text color when the countdown has a warning
+  color_warning_text = 'inherit',
+  -- Selector for the countdown element
```

Comment:
One last color-related thing: for now let's have these defaults be set by the CSS. I set up the CSS so that users can give a running/finished/warning background color and get reasonable text and border colors -- as long as those background colors as light enough.

In fact, now that I'm thinking about it and looking at this section, I think we might want to simplify the implementation a bit to not have `default_style`. The idea would be that all defaults live in the CSS so that the Quarto extension only needs to overlay user updates to those styles.

Reply:
@gadenbuie okay; this can be simplified up to just listing variable names. The filter will need a table of parameter names to parse through (c.f. `safeStyle()`). We may want to simplify `countdown_style()` as well since it will automatically populate with all the function defaults.

<img width="937" alt="Screenshot showing the outcome of running `countdown::countdown_style()`" src="https://github.com/gadenbuie/countdown/assets/833642/d251c74c-98c5-4868-a1c3-008c79e9aea5">

Reply:
Good point, thanks for bringing this up. I'll have to think about this, on the one hand I'm okay with `countdown_style()` in R having different behavior from the Quarto extension. On the other hand, I wasn't fully intending for every property to be set in every use of `countdown_style()`.